internal/wal_uploader.go

```diff
@@ -0,0 +1,61 @@
+package internal
+
+import (
+	"github.com/wal-g/storages/storage"
+	"github.com/wal-g/wal-g/internal/compression"
+	"github.com/wal-g/wal-g/utility"
+	"io"
+	"path"
+)
+
+// Uploader contains fields associated with uploading tarballs.
+// Multiple tarballs can share one uploader.
+type WalUploader struct {
```

Comment:
Just uploader is "wal uploader" too. DeltaFileManger is needed for use Wal-Delta. So it should be WalDeltaUploader.

I think you should have parent class or interface Uploader and 2 implementations WalUploader which doesn't use Wal-delta and WalDeltaUploader which use.

In the best case Uploader should be created once on initialization of wal-g  where it's possible  to choose implementation of Uploader in runtime depending on environment variable WALG_USE_WAL_DELTA(find it in config) and